scripts/ci/build-and-test.sh

```diff
@@ -2,6 +2,9 @@
 
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
+require_env_var PACT_BROKER_USERNAME
```

Comment:
Let me know if this works for env vars on this file for ci.

Reply:
Ah, also - you'll need to explicitly set these variables *after* `npm run dist`. The pact cores (ruby or rust) that pact-js uses configure themselves with `PACT_BROKER_PASSWORD` and `PACT_BROKER_USERNAME` - which means that if the variables are set before `npm run dist` then some of the tests may fail.

Reply:
`require_env_var` checks to see if a variable is set, but we'll want to actually set them with `export PACT_BROKER_PASSWORD=<whatever it is>`.

I forget the appropriate values, but they're hardcoded in the existing publish.js scripts. Even though they are credentials they are "safe" to be committed- they are for the public test account that's very kindly provided by Pactflow.

Reply:
> I forget the appropriate values, but they're hardcoded in the existing publish.js scripts. Even though they are credentials they are "safe" to be committed- they are for the public test account that's very kindly provided by Pactflow.
 
 This makes sense now, I was really hesitant to commit hard coded values. 😅 
 
 Thank you, I'll get these changes up today!